hal/src/rtl872x/ble_hal.cpp

```diff
@@ -2041,10 +2107,13 @@ void BleGap::handleDevStateChanged(T_GAP_DEV_STATE newState, uint16_t cause) {
     // NOTE: this event is generated before the connection is established
     if (newState.gap_adv_state != state_.state.gap_adv_state && newState.gap_adv_state == GAP_ADV_STATE_IDLE) {
         if (newState.gap_adv_sub_state == GAP_ADV_TO_IDLE_CAUSE_CONN) {
-            stopAdvertising(false);
-        } else if (advScheduled_) {
-            advScheduled_ = false;
-            startAdvertising(false);
+            isAdvertising_ = false; // adv timer has higher priority, just in case that the BLE_CMD_STOP_ADV_NOTIFY
+                                    // has been enqueued, followed by nontifying the ADV stopped event. Under
+                                    // certain circumstance notifying the ADV stopped event will cause BLE re-adv,
+                                    // e.g. in BLE listening mode.
+            if (os_timer_is_active(advTimeoutTimer_, nullptr)) {
+                os_timer_change(advTimeoutTimer_, OS_TIMER_CHANGE_STOP, hal_interrupt_is_isr() ? true : false, 0, 0, nullptr);
+            }
```

Comment:
When we  get here, the advertisement has been stopped by stack, we only need to stop the adv timer to prevent it from firing. The logic here is almost identical with operations in the `stopAdvertising()`, except for calling into `le_adv_stop()`. I'm just afraid that calling `le_adv_stop()` when stopped may mess up the adv state machine.